src/AdvancedHMC.jl

```diff
@@ -26,7 +26,7 @@ function Base.randn(rng::AbstractVector{<:AbstractRNG}, T, dim::Int, n_chains::I
     return cat(randn.(rng, T, dim)...; dims=2)
 end
 
-randcatlog(rng::AbstractRNG, unnorm_ℓp::AbstractVector{T}) where {T} =
+randcatlog(rng::AbstractRNG, unnorm_ℓp::AbstractVector) =
```

Comment:
`randcatlog` is not more readable than `randcat`... 
Maybe a docstring explaining that `randcat` assumes log probabilities is what we need here.

Reply:
I think this information should be reflected by the name. Basically `randcatlog` and `randcat` are two different functions. Even with a docstring it would make the code confusing when reading something like `randcat(ℓp)` but for `randcatlog(ℓp)` at least we would then realise what it is doing from the naming.

Reply:
Maybe there is a better name than `randcatlog`